lib/pos_blockchain.go

```diff
@@ -530,6 +533,17 @@ func (bc *Blockchain) canCommitGrandparent(currentBlock *BlockNode) (_grandparen
 	return nil, false
 }
 
+func (bc *Blockchain) setBlockCommittedStatus(blockNode *BlockNode, status CommittedBlockStatus) {
```

Comment:
Nit: pass in the minimum info needed to perform the work in a function. In this case, the function only needs the block hash
```suggestion
func (bc *Blockchain) setBlockCommittedStatus(blockHash *BlockHash, status CommittedBlockStatus) {
```